[칠면조] 2주차 TurkeyBook

dpudpu opened the pull request:
### 목록
- 친구 추가, 수락, 목록 조회 [프론트, 백엔드]
- 댓글, 답글 [프론트]
- 검색 및 추천  (결과는 아직 미구현)  [프론트, 백엔드]
- 파일 업로드 [백엔드]

pkch93 committed:
프로젝트 초기 세팅

pkch93 committed:
feat: BaseEntity 생성

pkch93 committed:
feat: 도메인 BaseEntity 상속

- Comment, Post, User 도메인에 BaseEntity 상속

pkch93 committed:
feat: template main layout 생성, 정적 컨텐츠 추가(css, images, js)

pkch93 committed:
refactor: junit4 어노테이션 junit5로 변경

pkch93 committed:
feat: 협업 베이스 코드 작성

Commit:
feat: User 도메인 생성

Commit:
feat: 유저 생성 기능 구현

- gradle: spring-boot-starter-webflux 추가

Commit:
feat: 유저 생성 검증 기능 구현

pkch93 committed:
feat: Post Entity 및 Contents Value Type 정의
- 커스텀 Exception 정의 (InvalidContentException, InvalidPostException)
- 테스트 코드 작성

Commit:
feat: 유저 조회 기능 구현

Commit:
refactor: UserRestController -> UserApiController 이름 변경

Commit:
refactor: 유저 조회, 생성 기능 Controller -> RestController 변경

pkch93 committed:
feat: PostService 생성기능 및 페이징 처리 조회기능
- Controller 및 Service 구현
- Request용 Dto인 PostRequest와 Response용 Dto인 PostResponse 구현
- 테스트는 아직 미완성

pkch93 committed:
test: PostService
생성 및 조회 테스트 구현 중 코드 프리징 (19.08.13)

Commit:
feat: 유저 삭제 기능 구현

Commit:
test: 유저 조회 테스트 작성

school0bhy committed:
refactor: User 연관 관계 제거

school0bhy committed:
refactor: Mock 테스트를 DataJpaTest로 변경

Commit:
test: 회원가입 폼 이동 테스트 작성

Commit:
test: UserService 테스트 작성

dpudpu committed:
feat: Comment 목록 조회

school0bhy committed:
settings: WebTestClient 의존성 추가

school0bhy committed:
feat: PostController 생성 로직 구현

- PostRequest @Valid로 값 검증
- 검증이 통과하지 못하는 경우 PostBadRequestException 발생
- Post 관련 에러를 처리하는 PostControllerAdvice 구현
- 테스트 구현

Commit:
refactor: ControllerAdvice basePackages -> basePackageClasses 변경

school0bhy committed:
refactor: PostConrollerAdvice 지정 패키지 속성 변경

basePackages -> basePackageClasses

school0bhy committed:
test: 페이지 조회

Commit:
refactor: 유저 DTO Lombok ToString 어노테이션 사용

Commit:
refactor: 테스트 코드 수정

- 중복 제거 (URI)
- 이메일 중복일 때 생성 에러 테스트 추가
- jsonPath -> 객체를 리턴받아 테스트

school0bhy committed:
feat: Post 수정 기능 구현

- Controller, Service, Post class에 관련 로직 구현
- PostResponse 기본 생성자 추가 (@NoArgConstructor)
- 테스트 추가

Commit:
refactor: User 검증 방식 수정

Commit:
refactor: Exception 수정

- 로그 추가
- SignUpException 메세지에 세부 사항 추가
- Exception 메세지 테스트 수정

school0bhy committed:
feat: Post 삭제 기능 구현

- Controller, Service, Post class에 관련 로직 구현
- 테스트 추가

Commit:
refactor: 컨트롤러 테스트 이름 수정

school0bhy committed:
refactor: PostBadRequestException 에러메시지 관리

skygl committed:
Merge pull request #13 from woowa-turkey/feature/user

회원 도메인 개발

school0bhy committed:
refactor: Post domain 관련 exception 에러메세지 관리

school0bhy committed:
refactor: NotExistPostException 에러메세지 관리

school0bhy committed:
styles: 코드 포맷팅

school0bhy committed:
refactor: PostService 테스트 코드 중복 제거

dpudpu committed:
feat: Comment 등록

school0bhy committed:
merge: User 도메인 / Post 도메인

school0bhy committed:
Merge branch 'master' into feature/post

dpudpu committed:
feat: Comment 삭제

dpudpu committed:
feat: Comment 삭제

dpudpu committed:
refactor: CommentService 메소드들 파라미터 수정

dpudpu committed:
Merge remote-tracking branch 'origin/develop' into feature/comment

dpudpu committed:
refactor: CommentApiController.list direction = Sort.Direction.ASC 파라미터 제거

yk1028 committed:
feat: login 기능 구현

dpudpu committed:
refactor: Comment.contents  @Column(nullable = false) 추가

dpudpu committed:
feat: CommentApiControllerAdvice.class

dpudpu committed:
delete: CommentController

pkch93 committed:
feat: Logging을 위한 AOP 적용

pkch93 committed:
fix: JPA Audit 적용 안되는 문제 해결
- @EnableJpaAuditing 추가

pkch93 committed:
feat: Post 생성, 조회 프론트 뷰
- ajax로 생성 로직 완료
- 첫 로딩시 10개의 포스팅 로딩 완료
- 스크롤을 90% 정도 내렸을 시 다음 10개의 포스팅 추가 구현 완료

school0bhy committed:
refactor: PostService에 @Transactional 적용

dpudpu committed:
test: CommentTest 추가

dpudpu committed:
refactor: Comment  @ForeignKey(name = "fk_comment_to_comment")) 추가

dpudpu committed:
refactor: CommentResponse  @JsonDeserialize(using = LocalDateTimeDeserializer.class) 제거

school0bhy committed:
feat: Post 수정, 삭제 프론트 뷰

school0bhy committed:
refactor: NotExistPostException -> NotFoundPostException 변경

school0bhy committed:
refactor: PostControllerAdvice 피드백 반영
- RestControllerAdvice
- ResponseEntity 팩토리 메서드 사용

school0bhy committed:
refactor: PostApiController
- ResponseEntity 팩토리 메서드 사용

school0bhy committed:
refactor: Post 도메인 피드백 반영
- update 메서드 파라미터 이름 통일 (other)

yk1028 committed:
refactor: Advice 수정

- 클래스명 수정 Exception 제거
- @ControllerAdvice -> @RestControllerAdvice

school0bhy committed:
refactor: service 피드백 반영
- PostService @Transactional 추가
- PostResponse @Builder 반영

yk1028 committed:
refactor: Service에 @Transactional 추가

dpudpu committed:
refactor: 댓글 삭제 flag 형식으로 변경

school0bhy committed:
refactor: test 코드 피드백 반영

- ErrorMessage 기본 생성자 추가 (@NoArgsConstructor)

school0bhy committed:
Merge branch 'feature/post-front' into feature/post

school0bhy committed:
refactor: error 메세지 출력

dpudpu committed:
refactor : CommentResponse 생성자 변경

기경 전 : private CommentResponse(final Comment comment)
변경 후 : private CommentResponse(final Long id, final String contents ...)

yk1028 committed:
feat: 로그아웃 기능 구현

yk1028 committed:
feat: User id 비교 메서드 추가

school0bhy committed:
feat: Exception 로그 추가

yk1028 committed:
feat: 유저 삭제 시 사용자 인증 추가 및 테스트

yk1028 committed:
refactor: 파라미터 이름 수정

yk1028 committed:
refactor: ControllerAdvice에서 리턴 방식 변경

yk1028 committed:
refactor: UserSessionHandlerMethodArgumentResolver -> UserSessionArgumentResolver 이름 변경

yk1028 committed:
style: 공백 제거

yk1028 committed:
refactor: UserSession 생성자 private로 변경

yk1028 committed:
refactor: UserApiController status code 변경

yk1028 committed:
Merge pull request #21 from woowa-turkey/feature/login

로그인, 로그아웃 개발

school0bhy committed:
Merge branch 'develop' into feature/post

school0bhy committed:
Merge pull request #14 from school0bhy/feature/post

 [게시글 도메인 개발] 철시, 올라프 미션 제출합니다.

pkch93 committed:
fix: Error시 200 OK 응답 문제 해결
- Exception logging시 void 리턴 문제

pkch93 committed:
fix: 더미데이터 제거 (ApplicationRunner 제거)

pkch93 committed:
refactor: @EnableJpaAuditing 위치 변경

pkch93 committed:
refactor: 존재하지 않는 값 명시화

school0bhy committed:
Merge pull request #22 from pkch93/bugfix

테스트 깨지는 문제 해결했습니다

pkch93 committed:
feat: Post 엔티티 User 필드 추가 (author)

dpudpu committed:
Merge branch 'develop' into feature/comment

pkch93 committed:
feat: Post 도메인 builder 추가 및 작성자 여부 기능 구현
- Post 도메인 테스트 추가 구현

pkch93 committed:
feat: PostService User 추가 구현
- 커스텀 Exception 추가 (NotPostOwnerException)
- 테스트 코드 추가

pkch93 committed:
refactor: Message 상수 이름 변경
- EXIST -> FOUND

pkch93 committed:
feat: PostApiController 유저 추가 구현
- 테스트 코드 구현

pkch93 committed:
feat: PostResponse에 author 필드 추가
- 게시글 템플릿에 작성자 표시되게 변경

school0bhy committed:
Merge pull request #23 from pkch93/feature/post

Post에 유저 매핑 기능 완료했습니다.

dpudpu committed:
refactor: CommentApiControllerAdvice  @ExceptionHandler 변경
변경 전 - @ExceptionHandler({RuntimeException.class})
변경 후 - @ExceptionHandler({ContextNotEmptyException.class, CommentSaveException.class, CommentAuthException.class})

dpudpu committed:
refactor: BaseControllerTests JSessionId 상수 처리

dpudpu committed:
feat: Comment - User, Post 매핑 완료

Commit:
feat: 회원 가입 프론트엔드 연동

Commit:
feat: 자바스크립트 공통 함수 분리

dpudpu committed:
feat: 댓글만, 답글만 조회 구현

dpudpu committed:
Merge branch 'develop' into feature/comment

Commit:
feat: 로그인 프론트엔드 연동

dpudpu committed:
fix: Comment 조회할 때 발생하는 에러 해결

Post 기본 생성자의 접근제어자가 private 이여서 프록시 객체가 안 되었음.
Private -> public 변경

dpudpu committed:
feat: BaseControllerTests 공통적으로 사용되는 값들 상수 생성

dpudpu committed:
refactor: LoggingAspect 추가로 인한 CommentApiController 의 log 제거

Commit:
refactor: favicon 변경

dpudpu committed:
refactor: BaseEntity setId() 추가

service 에서 Mock 으로 단위 테스트 하려면 필요함.
Pivotal 에서도 테스트를 위해서 setId() 만듬.

dpudpu committed:
test: CommentServiceTest

skygl committed:
Merge pull request #25 from woowa-turkey/feature/signupfront

회원가입/로그인 프론트엔드

dpudpu committed:
rename: CommentAuthException -> NotCommentOwnerException

dpudpu committed:
feat: 삭제된 댓글 삭제, 수정 시도 예외처리 추가

dpudpu committed:
fix: CommentServiceTest 댓글저장 테스트 문제 수정

dpudpu committed:
refactor: 모든 Exception log 제거

dpudpu committed:
feat: Comment getParentId() 추가

dpudpu committed:
refactor: CommentServiceTest 테스트메소드명 한글 -> 영어로 수정

dpudpu committed:
feat: Comment.update() Null 예외처리, CommentUpdateFailException 추가

dpudpu committed:
feat: Comment.contents 공백, null 예외처리 추가

dpudpu committed:
style: 포맷팅

dpudpu committed:
Merge pull request #27 from woowa-turkey/feature/comment

[백엔드] 댓글 도메인 개발

dpudpu committed:
Merge pull request #29 from woowa-turkey/feature/comment

[Fix] 댓글 push 안 한 부분 pr 후 머지

dpudpu committed:
refactor: NotPostOwnerException 메시지 변경

dpudpu committed:
style: build.gradle 정리

dpudpu committed:
refactor: ErrorMessage commons.ErrorMessage 로 통일

dpudpu committed:
refactor: BaseControllerTests jsessionId, Email, password, name 상수 처리

dpudpu committed:
rename: ResolverConfig -> WebConfig

yk1028 committed:
feat: 배포 스크립트 작성 및 build.gradle 수정

yk1028 committed:
refactor: 배포 스크립트 수정

yk1028 committed:
Merge pull request #30 from woowa-turkey/feature/deploy

배포 준비

school0bhy committed:
refactor: ResponseEntity 반환 컨벤션 통일
- Created 상태코드의 URI 값 추가

school0bhy committed:
refactor: PostApiController 위치 변경

dpudpu committed:
refactor: CommentRepository.findAllByPostId() JPQL 없이 조회하게 수정

findAllByPostId -> findAllByPostIdAndParentIdIsNull

dpudpu committed:
refactor: Comment.update()

this.contents = other.getContents() -> other.contents

dpudpu committed:
refactor: Comment.getParent(), getParentCommentId() return Optional 로 변경

school0bhy committed:
refactor: PostApiController의 BindingResult 처리 부분 중복 제거
- PostControllerAdvice에서 BindException을 처리하도록 함

school0bhy committed:
Merge pull request #39 from woowa-turkey/refactor/post

게시글 피드백 반영

dpudpu committed:
Merge pull request #41 from woowa-turkey/refactor/Comment

댓글 피드백 반영

pkch93 committed:
refactor: LoggingAspect exception pointcut 수정

pkch93 committed:
refactor: onload 제거

Commit:
feat: 프론트엔드 로그아웃 기능 추가

Commit:
refactor: javascript var 변수 피드백 반영

- var 대신 let 사용

Commit:
refactor: 회원 가입 form의 input 태그 아이디 값 수정

Commit:
refactor: 서버와 jquery ajax로 통신하도록 변경

- fetch API 에서 jquery ajax 사용

skygl committed:
Merge pull request #42 from pkch93/refactor/post

Post 프론트 피드백 반영

Commit:
Merge remote-tracking branch 'origin/develop' into feature/logoutfront

skygl committed:
Merge pull request #43 from woowa-turkey/feature/logoutfront

로그아웃 버튼 추가 및 프론트 피드백 반영

pkch93 committed:
feat: LoginInterceptor 구현 및 적용
- LoginInterceptor 적용에 따른 테스트 코드 수정

pkch93 committed:
refactor: 도메인 NotFound Exception을 기존의 javax.persistence.EntityNotFoundException으로 변경

pkch93 committed:
refactor: BaseEntity Column 명세 작성

pkch93 committed:
refactor: app.js, api.js 폴더링 변경

pkch93 committed:
feat: 기본 더미데이터 추가 (data.sql)

pkch93 committed:
refactor: 공통 MediaType 상수화

pkch93 committed:
styles: 소스 코드 포메팅

pkch93 committed:
fix: 글 작성시 개행 적용 안되는 버그 해결

pkch93 committed:
settings: lite-server
- front 작업을 쉽게 하기 위한 lite-server 추가
- npm run dev로 lite-server 실행 설정

pkch93 committed:
refactor: main layout 공통으로 사용할 수 있도록 일부 수정
- main layout 수정에 따른 index.html 수정

pkch93 committed:
refactor: 공통 css 파일인 shared.css 생성
- head.html 추가

pkch93 committed:
refactor: 정적 파일 경로 수정

pkch93 committed:
feat: mypage front 구현

dpudpu committed:
feat: 댓글, 대댓글 프론트 CRUD

dpudpu committed:
refactor: console.log 제거

dpudpu committed:
fix: 대댓글 수정 완료 후 이상하게 보이는 부분 수정

dpudpu committed:
feat: 이름으로 유저 검색

dpudpu committed:
feat: 이름으로 검색 프론트

dpudpu committed:
fix: 댓글 수정폼 클릭시 답글도 수정폼 나오는 문제 해결

dpudpu committed:
Merge pull request #46 from pkch93/feature/feed-page

개인 피드 페이지

dpudpu committed:
Merge pull request #47 from woowa-turkey/feature/댓글_프론트

댓글 프론트

dpudpu committed:
Merge branch 'develop' into feature/search

dpudpu committed:
Merge pull request #50 from woowa-turkey/feature/search

이름으로 검색

dpudpu committed:
Merge pull request #51 from woowa-turkey/feature/댓글_프론트

fix: 댓글 수정폼 클릭시 답글도 수정폼 나오는 문제 해결

dpudpu committed:
fix: 댓글 개행 문제 해결

dpudpu committed:
Merge pull request #52 from woowa-turkey/fix/comment_newLine

fix: 댓글 개행 문제 해결

dpudpu committed:
refactor: 댓글 수정 폼 변경

dpudpu committed:
feat: 댓글 시간 표시

skygl committed:
Merge pull request #54 from woowa-turkey/feature/comment_time

[프론트] 댓글 시간 표시

pkch93 committed:
Revert "[칠면조] 1주차 Facebook 기능 구현 (#8)"

This reverts commit f10e4baa82f5d6b3179f3efaf2b97f7b3679f5ff.

pkch93 committed:
Merge branch 'master' into develop

skygl committed:
친구 맺기 (#55)

* feat: 친구 추가 프론트

* feat: 친구 요청 기능 추가

* feat: 받은 친구 요청 조회 기능 추가

* feat: 친구 요청 기능 프론트 구현

* feat: 친구 요청 중복 예외처리

* refactor: 오른쪽 유저 목록에서 자신을 제외하도록 수정

* feat: 친구 요청 수락 기능 구현

* feat: 친구 조회 기능 추가

- 친구 맵핑 단방향에서 양방향으로 수정

* feat: 친구 목록 조회 프론트 추가

* feat: 삭제 버튼 변경

* refactor: FriendRequest -> FriendAsk 이름 변경

* refactor: 주석 제거

* refactor: Friend 생성 메서드 수정

* refactor: 테스트 리팩토링

* refactor: FriendApiController Created URI null로 수정

school0bhy committed:
게시글 파일 첨부 기능 백엔드 (#56)

* settings: S3 관련 의존성 및 설정 추가

* feat: 파일 저장 기능 구현

* feat: 게시글 파일 첨부 기능 구현
- 다중 파일 가능
- 게시글 작성 시에만 가능, 게시글 수정 시에는 불가

dpudpu committed:
test: PostServiceTest 깨지는 테스트 수정

dpudpu committed:
refactor: build.gradle 정리

dpudpu committed:
refactor: Response 생성자 -> 팩터리 메서드로 변경

yk1028 committed:
test: given, when, then 추가

dpudpu committed:
refactor: BaseEntity에서 UpdatableEntity 분리

dpudpu committed:
fix: 게시글 추가 안되는 문제 해결

dpudpu committed:
Merge pull request #57 from woowa-turkey/refactor/all

합친 후 생긴 문제 해결

wmakerjun commented on src/main/resources/static/js/friendask/load.js:
```diff
@@ -0,0 +1,77 @@
+$(document).ready(function() {
```
$(document).ready()는 body태그의 모든 태그들이 출력된 다음에 호출이 되는 코드입니다.
그런데 load.js는 다른 html 페이지에서 body 태그가 닫히기 전에 선언되어 있기 때문에, 이미 DOM 태그들이 출력된 다음 load 되므로, 굳이 $(document).ready() 안에서 init function을 실행시킬 필요가 없어보입니다.

wmakerjun commented on src/main/resources/static/js/friendask/load.js:
```diff
@@ -0,0 +1,77 @@
+	friendAskModule.init()
+})
+
+let friendAskModule = (function() {
```
friendAskModule이 let으로 선언된 이유는 무엇일까요? let과 const의 차이를 확인해본 후 type에 맞게 변경해주면 좋을 것 같아요 :)

wmakerjun commented on src/main/resources/static/js/friendbar/load.js:
```diff
@@ -0,0 +1,94 @@
+
+let friendBarModule = (function() {
+	function loadAllUser() {
+	    let res = $.ajax({
```
1. res를 let으로 선언할 필요가 없어 보입니다.
2. api.js에서 fetch를 이용해서 비동기통신을 하기 위한 객체를 만들었는데, 여기선 jquery를 이용해 비동기 통신을 진행하고 있습니다. 한 웹서비스를 만들 때 비동기 통신하는 방식은 한가지로 통일해야 합니다.

wmakerjun commented on src/main/resources/static/js/friendbar/load.js:
```diff
@@ -0,0 +1,94 @@
+	        dataType:'json'
+	    })
+
+	    let data = res.responseJSON
```
data를 let으로 선언할 필요가 없어 보입니다.

wmakerjun commented on src/main/resources/static/js/friendbar/load.js:
```diff
@@ -0,0 +1,94 @@
+	friendBarModule.init()
+})
+
+let friendBarModule = (function() {
```
모듈을 let으로 선언한 이유가 있을까요?

wmakerjun commented on src/main/resources/static/js/friendbar/load.js:
```diff
@@ -0,0 +1,94 @@
+	}
+
+	function loadFriends() {
+	    let res = $.ajax({
```
let으로 선언할 필요가 없어보입니다.

wmakerjun commented on src/main/resources/static/js/friendbar/load.js:
```diff
@@ -0,0 +1,94 @@
+            dataType:'json'
+        })
+
+        let data = res.responseJSON
```
let으로 선언할 필요가 없어보입니다.

wmakerjun commented on src/main/resources/static/js/friendask/load.js:
```diff
@@ -0,0 +1,77 @@
+			btns[i].addEventListener('click', function() {
+			    let data = {"friendAskId" : btns[i].parentNode.getAttribute("data-id")}
+
+	            let res = $.ajax({
```
let으로 선언할 필요가 없어보입니다.

wmakerjun commented on src/main/resources/static/js/friendask/load.js:
```diff
@@ -0,0 +1,77 @@
+		let btns = document.getElementsByClassName('friend-ask-accept-btn')
+		for(let i = 0; i < btns.length; i++) {
+			btns[i].addEventListener('click', function() {
+			    let data = {"friendAskId" : btns[i].parentNode.getAttribute("data-id")}
```
let으로 선언할 필요가 없어보입니다.

wmakerjun reviewed (changes requested):
전반적으로 프론트엔드 코드를 모듈화하고, 역할과 책임을 분리한 점이 멋집니다 👍 
다만 변수를 선언할 때 let과 const의 사용이 아직 익숙치 않아 실수들이 조금 발견되었습니다.
변수가 let인지 const인지 잘 구분해주길 부탁해요 😉

pkch93 committed:
글 / 댓글 좋아요 백엔드 (#58)

* feat: Good domain 구현

* feat: PostGood domain 구현
- Good 기본 엔티티를 상속받는 PostGood 구현
- PostGoodRepository 구현 + 테스트 작성

* feat: PostGood Service 단 구현
- PostGood 로직을 담당하는 PostGoodService 구현
- PostGoodService와 PostService 간의 의존관계 생성
- Service test 작성

* feat: PostApiController에 PostGood 관련 url 매핑

* refactor: PostGood @Builder 제거

* refactor: Good 추상클래스에 유저정보만 가지도록 수정
- 그 외 필드는 각 도메인 Good 클래스에서 가지도록 수정

* refactor: 좋아요 Response용 객체 GoodResponse 생성

* feat: CommentGood 도메인 구현
- Repository 구현
- 테스트 구현

* feat: CommentGood 도메인 로직을 담당하는 CommentGoodService 구현

* feat: CommentService에서 CommentGoodService를 가지고 좋아요 기능 구현
- 테스트 추가

* feat: 댓글 좋아요 컨트롤러 경로 추가
- 테스트 추가

school0bhy committed:
게시글 첨부파일 리팩터링 적용 (#60)

* refactor: 메서드 파라미터에 final 추가

* refactor: 개행 등 코드 정리

* refactor: fileentity -> uploadfile 적용 안 돼있던 부분 적용

* refactor: FileDto -> FileResponse로 이름 변경

* refactor: S3Connector
- bucket 명 AwsConfig 파일 활용해 주입 받기
- 디렉토리명, 파일명 각각 전달받도록 변경

* refactor: PostRequest 생성자 대신 builder 활용

school0bhy committed:
게시글 작성 이미지 첨부 프론트 (#62)

* refactor: 메서드 파라미터에 final 추가

* refactor: 개행 등 코드 정리

* refactor: fileentity -> uploadfile 적용 안 돼있던 부분 적용

* refactor: FileDto -> FileResponse로 이름 변경

* refactor: S3Connector
- bucket 명 AwsConfig 파일 활용해 주입 받기
- 디렉토리명, 파일명 각각 전달받도록 변경

* refactor: PostRequest 생성자 대신 builder 활용

* feat: 파일 첨부하여 게시물 작성하기 프론트 구현

* feat: 이미지가 포함된 게시물 뷰 렌더링 적용

* refactor: 게시물 버튼 관련 js 에러 처리
- null체크
- hasClass 함수를 먼저 호출해 classList 호출 시 에러 발생 제거

yk1028 committed:
친구 삭제 & 예외처리  (#63)

* feat: 친구 요청 거절 기능 구현

* feat: 친구 삭제 기능 구현

* feat: 상대가 친구 요청을 보낸 경우 친구 요청 불가능하도록 예외 처리

* feat: 이미 친구인 경우 친구 요청 불가능하도록 예외 처리

* feat: 유저 삭제시 관련된 친구, 친구 요청 삭제

* test: 접근 제어자 수정

pkch93 committed:
좋아요 기능 (#64)

* style: 코드 포메팅

* refactor: PostGood 응답 로그인한 사용자가 해당 Post 좋아요 했는지 포함
하도록 변경
- PostGoodService 메서드 분리
- PostGoodRepository countByPost -> findByPost
- Good 클래스에 해당 사용자가 좋아요 표시를 했는지 보여주는 isGood 구현

* feat: PostGood 프론트 작업 완료

* refactor: CommentGood

pkch93 committed:
[mypage 연결] (#65)

* feat: Mypage 연결 핸들러 메서드 생성

* feat: mypage

dpudpu committed:
게시글 수정, 삭제 안되는 문제 및 댓글 에러 처리 (#66)

* fix: 게시글 수정, 삭제 안되는 버그 해결

* fix: PostServiceTest 깨지는 테스트 수정

dpudpu committed:
검색 결과 구현 (#67)

* feat: 검색 결과 기본 폼 구현

* fix: 전체검색 결과 여러개 나오는 문제 해결

* refactor: User 이름으로 검색 Page로 받게 변경

* feat: 유저 검색 결과 구

dpudpu committed:
fix: SearchController 뷰네임 / 삭제

woowahan-pjs commented on src/main/java/com/wootecobook/turkey/friend/controller/api/FriendApiController.java:
```diff
@@ -0,0 +1,62 @@
+package com.wootecobook.turkey.friend.controller.api;
```
`controller.api` 패키지까지 필요할까요?

dpudpu replied:
controller 패키지에 RestController와 Controller를 분리하기 위해서 패키지를 나누었는데, 패키지 구조를 레이어가 아닌 모듈 기준으로 나눠서 Controller가 많지 않아서 클래스명으로도 충분히 구분이 된다고 판단해서 다시 합쳤습니다.

woowahan-pjs commented on src/main/java/com/wootecobook/turkey/friend/controller/api/FriendApiController.java:
```diff
@@ -0,0 +1,62 @@
+@RequestMapping("/api/friends")
+public class FriendApiController {
+
+    private FriendAskService friendAskService;
```
`final`을 사용하지 않은 이유가 있나요?

```suggestion
    private final FriendAskService friendAskService;
```

dpudpu replied:
실수했습니다. 반영했습니다. 감사합니다. 👍

woowahan-pjs commented on src/main/java/com/wootecobook/turkey/comment/controller/api/CommentApiController.java:
```diff
@@ -69,5 +70,19 @@ public ResponseEntity delete(@PathVariable Long id,
     }
+
+    @GetMapping("/{id}/good")
+    public ResponseEntity<GoodResponse> good(@PathVariable Long id, UserSession userSession) {
```
어노테이션 기반으로 동작할 수 있게 구현하면 어떤 이점이 있을까요?

```suggestion
    public ResponseEntity<GoodResponse> good(@PathVariable Long id, @LoginUser UserSession userSession) {
```

dpudpu replied:
명시적으로 로그인된 유저임을 확인할 수 있었습니다.
혹시 또 다른 이점이 있을까요?

dpudpu committed:
feat: 이스터에그 추가

woowahan-pjs commented on src/main/java/com/wootecobook/turkey/comment/domain/Comment.java:
```diff
@@ -13,12 +13,14 @@
 @Getter
 @NoArgsConstructor
-public class Comment extends BaseEntity {
+public class Comment extends UpdatableEntity {
```
`isDeleted`라는 변수 이름에 대해서 이야기하고 싶네요.
https://stackoverflow.com/questions/5322648/for-a-boolean-field-what-is-the-naming-convention-for-its-getter-setter

dpudpu replied:
Deleted로 수정했습니다. 감사합니다.

woowahan-pjs commented on src/main/java/com/wootecobook/turkey/comment/domain/Comment.java:
```diff
@@ -13,12 +13,14 @@
 @Getter
 @NoArgsConstructor
-public class Comment extends BaseEntity {
+public class Comment extends UpdatableEntity {
```
`@where`에 대해서도 알아보면 좋겠어요.

dpudpu replied:
반영했습니다. 감사합니다.

woowahan-pjs commented on src/main/java/com/wootecobook/turkey/comment/service/CommentGoodService.java:
```diff
@@ -0,0 +1,44 @@
+        Optional<CommentGood> maybeCommentGood = commentGoodRepository.findByCommentAndUser(comment, user);
+
+        if (maybeCommentGood.isPresent()) {
+            cencelGoodRequest(maybeCommentGood.get());
```
'cencel'? 'cancel'!

dpudpu replied:
반영했습니다. 감사합니다.

woowahan-pjs commented on src/main/resources/templates/mypage.html:
```diff
@@ -0,0 +1,70 @@
+        <!-- 프로필 사진 및 배경 구역 feed-header -->
+        <header class="mypage-header">
+            <div class="background pointer">
+                <span class="username" style="color: black">박경철 (철시)</span>
```
이스터 에그

dpudpu replied:
들켰습니다..ㅠ

woowahan-pjs commented on src/main/java/com/wootecobook/turkey/comment/domain/Comment.java:
```diff
@@ -13,12 +13,14 @@
 
 @Entity
 @Getter
 @NoArgsConstructor
```
`AccessLevel`을 사용하지 않은 이유가 있나요?

dpudpu replied:
일부는 레이지로딩 할 때 프록시 객체 생성하려면 기본 생성자가 필요해서 public으로 오픈했고 
일부는 깜박했습니다. 감사합니다.

woowahan-pjs reviewed (changes requested):
전체적으로 접근 제어자에 대한 점검이 필요해 보여요.
`@NoArgsConstructor`, `@Builder` 등을 사용하고 있는 곳을 확인해 주세요.

dpudpu committed:
refactor: Controller 패키지 구조 변경

dpudpu committed:
fix: signup.css 에러 처리

dpudpu committed:
refactor: 필드에 final 추가

dpudpu committed:
refactor: rename Comment.isDeleted -> deleted

dpudpu committed:
refactor: Comment @Where 추가

dpudpu committed:
refactor: Entity 기본 생성자 private 접근제어자 추가

dpudpu committed:
feat: @LoginUser 추가

dpudpu committed:
style: 포맷팅

dpudpu commented:
> 전체적으로 접근 제어자에 대한 점검이 필요해 보여요.
> `@NoArgsConstructor`, `@Builder` 등을 사용하고 있는 곳을 확인해 주세요.

피드백 주신 부분 반영했습니다. 감사합니다. 
`@Builder`가 있는데 접근제어자를 public으로 준 이유는 
테스트 코드 작성할 때 인스턴스 생성시 builder로 작성하면 인스턴스가 1~2개면 괜찮지만 여러 개일 경우에 너무 길어져서 생성자로 인스턴스를 생성하려고 했습니다. 혹시 이 부분에 대해서 어떻게 생각하시나요?

Commit:
refactor: 프론트 피드백 수정

- let -> const 수정
- jQuery ajax 에서 api.js의 Fetch API 사용

dpudpu committed:
feat: GoodService interface (#73)

dpudpu committed:
feat: GoodService

school0bhy committed:
첨부파일 이미지 프리뷰 프론트입니다! (#74)

* refactor: 파일첨부 event 캡쳐링 적용
- document.getElementsByClassName -> document.getElementById

* feat: 첨부파일 프리뷰 프론트 구현

* feat: 게시글 저장후 첨부파일 프리뷰 이미지 비우기

school0bhy committed:
fix: 배포 스크립트 수정
